### ISSUE TITLE
Plugins for Frontend

### ISSUE THREAD

**Issue by kasvith** [joomla-projects/media-manager-improvement#382]
It should be able for thirdparty app developers to extend current capability of the frontend via plugins. For example think of a plugin developer want to provide a *Custom Action* called "Compress and Email..." which can be executed on any directory or file. 

So we need a mechanism to extend current media manager frontend in such cases.
This makes Media Manager a very powerful Filemanager out there.

**Comment by dgrammatiko** [comment]
Ok, can you give me an example of how such a `compress and email` functionality should work?
And most importantly on which context ( I guess input field).?

**Comment by kasvith** [comment]
Nope, i mean we should let 3rd party developers extend the functionality of the mediamanager also in fronted too

For example think when you install winRAR on Windows it adds a custom action to context menu. 

If we have something like that, mediamanager will be more powerful 

For the task we need a modular system in frontend

**Comment by laoneo** [comment]
Honestly I like the idea, but it would mean we start to introduce a similar plugin system as we have for the PHP part. Where extension devs can hook into different actions within media manager.

**Comment by kasvith** [comment]
Yes, this is for js side, we will need some php actions too, with this mediamanager will be a pro filemanager among other cms

**Comment by dneukirchen** [comment]
I like the idea. Have to do some research on this. Might be something that i can check out when we implement #351, but i would not put too much time into this atm because we have a lot of other stuff to do for the current milestones.

**Comment by kasvith** [comment]
For example we need to take our actions like Rename as a modular plugin, 

With this 3dps can create filesystem plugins with their unique actions

**Comment by kasvith** [comment]
Yup we can work on this later

**Comment by laoneo** [comment]
Probably something the JS team can think about in general as it can be that we will need it in rest of Joomla on some day as well. Just an idea.

**Comment by dneukirchen** [comment]
@laoneo https://volunteers.joomla.org/teams/javascript-working-group#getinvolved 🙄
